sycl/source/detail/allowlist.cpp

```diff
@@ -239,10 +239,12 @@ AllowListParsedT parseAllowList(const std::string &AllowListRaw) {
 // Checking if we can allow device with device description DeviceDesc
 bool deviceIsAllowed(const DeviceDescT &DeviceDesc,
                      const AllowListParsedT &AllowListParsed) {
-  for (const auto &SupportedKeyName : SupportedAllowListKeyNames)
+  for (const auto &SupportedKeyName : SupportedAllowListKeyNames) {
+    (void)SupportedKeyName;
     assert((DeviceDesc.find(SupportedKeyName) != DeviceDesc.end()) &&
            "DeviceDesc map should have all supported keys for "
            "SYCL_DEVICE_ALLOWLIST.");
+  }
```

Comment:
```suggestion
  assert(std::all_of(SupportedAllowListKeyNames.begin(),
                     SupportedAllowListKeyNames.end(),
                     [&DeviceDesc](const auto &SupportedKeyName) {
                       return DeviceDesc.find(SupportedKeyName) !=
                              DeviceDesc.end();
                     }) &&
         "DeviceDesc map should have all supported keys for "
         "SYCL_DEVICE_ALLOWLIST.");
```
Let's use standard library instead of a raw loop.
Don't forget to add `#include <algorithm>`.

Reply:
Done

Reply:
Can you guess what you forgot to do? :)